### DISCUSSION TITLE
showing a title or message with the showBubble

### DISCUSSION THREAD

**Discussion by fahimfahadleon** [dofire/Floating-Bubble-View#24] [Ideas]
I was wondering if there is a way to show not only the icon but also a message such as you have a new message or the message itself. It would be helpful. Like messenger does.

**Comment by dofire** [comment]
At the moment, our library doesn't have the capability to display a message along with the icon. It's an interesting idea though, and I'll keep it in mind for a future update if it's possible. 🍀
Thank you for the suggestion. ❤️

**Reply by fahimfahadleon** [reply]
Sir how can i detect back press while expendable view is visible? I want to switch to bubble view on backpress.

**Reply by dofire** [reply]
Hi @fahimfahadleon,

In order to detect back press while the expandable-view is visible, you can add the following code before inflating the layout. Here's an example in Kotlin:
```kotlin
        ...
        val wrapper: ViewGroup = object : FrameLayout(this) {

            override fun dispatchKeyEvent(event: KeyEvent): Boolean {
                if (event.keyCode == KeyEvent.KEYCODE_BACK) {
                    action.popToBubble()
                    return true
                }
                return super.dispatchKeyEvent(event)
            }

        }

        val layout = inflater.inflate(R.layout.layout_view_test, wrapper) // pass "wrapper" here

        layout.findViewById<View>(R.id.card_view).setOnClickListener { v: View? ->
            Toast.makeText(this, "hello from card view from kotlin", Toast.LENGTH_SHORT).show();
            action.popToBubble()
        }
        ...
```

**Reply by fahimfahadleon** [reply]
saved my day sir. however 
```
layout.findViewById<View>(R.id.card_view).setOnClickListener { v: View? ->
            Toast.makeText(this, "hello from card view from kotlin", Toast.LENGTH_SHORT).show();
            action.popToBubble()
        } 
```
line of code is for the functionality of my view right? such as 
```
@Nullable
    @Override
    public ExpandableView.Builder setupExpandableView(@NonNull ExpandableView.Action action) {
        Log.e("chatBubbleService", "setupExpandableView");
        firstAction = action;
        LayoutInflater inflater = (LayoutInflater) getSystemService(LAYOUT_INFLATER_SERVICE);

        ViewGroup viewGroup = new FrameLayout(context){//I have doubt here because the parent of my layout is ConstrainLayout
//anyway this isn't serious issue i just copied what you  said and pasted it here. :D
            @Override
            public boolean dispatchKeyEvent(KeyEvent event) {
                if (event.getKeyCode() == KeyEvent.KEYCODE_BACK) {
                    action.popToBubble();
                    return true;
                }
                return super.dispatchKeyEvent(event);
            }
        };

        View layout = inflater.inflate(R.layout.forground_service_container, viewGroup);
         linearLayout = layout.findViewById(R.id.itemContainer);

        if (data.optInt("action") == PushNotificationService.SEND_MESSAGE) {
//registering view buttons and layouts clicks
            vi = getView(inflater, action);
        } else if (data.optString("action").equals(String.valueOf(PushNotificationService.SEND_AUDIO_CALL))) {
//registering view buttons and layouts clicks
            vi = getCallView(inflater, PushNotificationService.SEND_AUDIO_CALL, action);
        } else if (data.optString("action").equals(String.valueOf(PushNotificationService.SEND_VIDEO_CALL))) {
//registering view buttons and layouts clicks
            vi = getCallView(inflater, PushNotificationService.SEND_VIDEO_CALL, action);
        }

        linearLayout.addView(vi);

        return new ExpandableView.Builder(this)
                .expandableView(layout)
                .dimAmount(0.8f)
                .addExpandableViewListener(new ExpandableView.Listener() {
                    @Override
                    public void onOpenExpandableView() {
                    }

                    @Override
                    public void onCloseExpandableView() {
                    }
                })
                .expandableViewStyle(null);
    }
```

**Reply by dofire** [reply]
yes you're right 👍

**Reply by fahimfahadleon** [reply]
hello sir,
i am facing a problem. I have a xiaomi phone (POCO F3) which is running in android 12. The phone has gesture enabled so there is no soft back button or home button. But when expended view is shown the gesture is not working so i can't back from expandable view to bubble view. Can i get any help?

**Reply by dofire** [reply]
Hi @fahimfahadleon ✋,

I tested the back swipe-gesture on two devices:  Redmi Note 7 running AOSP Android 11 and Poco X3 Pro running MIUI Android 12. The gesture worked perfectly fine on the RMN7, but the X3 Pro faced the same issue as yours. In my opinion, this may be a problem with MIUI. However, I will look into this further and try to find a solution. Thank you for pointing me out.

**Reply by fahimfahadleo** [reply]
Please check if rotation works perfectly. My device seems to have some problems on rotation.

**Comment by fahimfahadleo** [comment]
sir it would be kind of you if you make some animation for poptobubble and navigate to expendableview.

**Reply by dofire** [reply]
Thank you. I'll take some time to review your suggestion and see how I can implement it.

**Comment by fahimfahadleo** [comment]
hello brother i can't find onDestroy() method in the service how do i get it and also i want to know if expandableview is visible or bubble is visible. can you help me?

**Reply by dofire** [reply]
I'm not sure what your issue is, but typically you can override the onDestroy() method in the service without any issues.
```
override fun onDestroy() {
      super.onDestroy()
      // your code here
}
```

As for detecting whether an expandable view or bubble is visible, you can use a static variable to keep track of the visibility status. For example, you can set a boolean flag to true when the view is expanded or the bubble is visible, and set it to false when it's not. This way, you can easily check the flag to determine whether the view/bubble is visible or not.

**Reply by fahimfahadleo** [reply]
thank you sir. but I can't find onDestroy method. I edited the library and added a medhod onServiceKilled and used it. but i think the other problem i said should be included in the library itself. its important when chat bubble is used in complex cases such as using it as messenger chat bubble.

**Reply by dofire** [reply]
Thank you, I'll consider it for the next update

**Reply by fahimfahadleo** [reply]
maybe you are wondering why i am asking so much question in a discussion? Let me show you what I am really doing. please enjoy. [screen record](https://drive.google.com/file/d/1SWPAmpZGwkFhnaM0qy9v3eRCD9oDy6sX/view?usp=sharing)

**Reply by dofire** [reply]
Thanks for sharing the screen record, it's really helpful to see what you're working on and how you're using the library. Don't worry about asking too much questions, that's what the discussion is for! Keep up the good work!
